core/src/main/scala/pureconfig/DurationConvert.scala

```diff
@@ -42,9 +43,11 @@ private[pureconfig] object DurationConvert {
    */
   def fromDuration(d: Duration): String = {
     d match {
+      case f: FiniteDuration => fromFiniteDuration(f)
+      // We must do an `eq` instead of `==` comparison because Undefined is intentionally != itself.
+      case i: Duration.Infinite if i eq Duration.Undefined => Duration.Undefined.toString
```

Comment:
Minor suggestion: the `Duration.Infinite` explicit typing in this whole `match` is redundant - you can just write `case d if d eq Duration.Undefined`, `case Duration.MinusInf` and `case Duration.Inf`. With that, you cover all three subclasses of the `Duration.Infinite` sealed trait.